src/router_core/delivery.c

```diff
@@ -712,14 +712,15 @@ static void qdr_delivery_anycast_update_CT(qdr_core_t *core, qdr_delivery_t *dlv
     //
     // Non-multicast Logic:
     //
-    // If disposition has changed and there is a peer link, set the disposition
-    // of the peer
+    // If the disposition or extended delivery-state has changed and there is a
+    // peer link, set the disposition/delivery-state of the peer
     // If remote settled, the delivery must be unlinked and freed.
     // If remote settled and there is a peer, the peer shall be settled and
     // unlinked.  It shall not be freed until the connection-side thread
     // settles the PN delivery.
     //
-    if (new_disp != dlv->remote_disposition) {
+    // kag fix me!
+    if (new_disp != dlv->remote_disposition || dlv->remote_state) {
```

Comment:
An alternative here would be to test the disposition. E.g. TCP_SEQ_ACK (or PN_RECEIVED) would imply an update required.

Reply:
Good point - I'm going to need to refactor this a bit more.  This POC is exposing a few assumptions we've made regarding disposition propagation that need to be revisited.